test/new-e2e/tests/agent-platform/common/agent_behaviour.go

```diff
@@ -319,3 +320,53 @@ func CheckCWSBehaviour(t *testing.T, client *TestClient) {
 		require.True(tt, result, "system-probe and security-agent should communicate")
 	})
 }
+
+// CheckSystemProbeBehavior runs tests to check the agent behave correctly when system-probe is enabled
+func CheckSystemProbeBehavior(t *testing.T, client *TestClient) {
+	t.Run("enable system-probe and restarts", func(tt *testing.T) {
+		err := client.SetConfig(client.Helper.GetConfigFolder()+"system-probe.yaml", "system_probe_config.enabled", "true")
+		require.NoError(tt, err)
+
+		err = client.SetConfig(client.Helper.GetConfigFolder()+"system-probe.yaml", "system_probe_config.enabled", "true")
+		require.NoError(tt, err)
+
+		_, err = client.SvcManager.Restart(client.Helper.GetServiceName())
+		require.NoError(tt, err, "datadog-agent should restart after CWS is enabled")
+	})
+
+	t.Run("system-probe is running", func(tt *testing.T) {
+		var err error
+		require.Eventually(tt, func() bool {
+			return AgentProcessIsRunning(client, "system-probe")
+		}, 1*time.Minute, 500*time.Millisecond, "system-probe should be running ", err)
+	})
+
+	t.Run("ebpf programs are unpacked and valid", func(tt *testing.T) {
+		ebpfPath := "/opt/datadog-agent/embedded/share/system-probe/ebpf"
+		output, err := client.Host.Execute(fmt.Sprintf("find %s -name '*.o'", ebpfPath))
+		require.NoError(tt, err)
+		files := strings.Split(strings.TrimSpace(output), "\n")
+		require.Greater(tt, len(files), 0, "ebpf object files should be present")
+
+		_, err = client.Host.Execute("command -v readelf")
+		if err != nil {
+			tt.Skip("readelf is not available on the host")
+			return
+		}
+
+		hostArch, err := client.Host.Execute("uname -m")
```

Comment:
We can use [uname](https://pkg.go.dev/golang.org/x/sys/unix#Uname) here directly.